pkg/daemon/update_test.go

```diff
@@ -318,6 +318,57 @@ func TestUpdateSSHKeys(t *testing.T) {
 	}
 }
 
+// This test should fail until Ignition validation enabled.
+// Ignition validation does not permit writing files to relative paths.
+func TestInvalidIgnConfig(t *testing.T) {
+	// expectedError is the error we will use when expecting an error to return
+	expectedError := fmt.Errorf("broken")
+	// testClient is the NodeUpdaterClient mock instance that will front
+	// calls to update the host.
+	testClient := RpmOstreeClientMock{
+		GetBootedOSImageURLReturns: []GetBootedOSImageURLReturn{},
+		RunPivotReturns: []error{
+			// First run will return no error
+			nil,
+			// Second rrun will return our expected error
+			expectedError},
+	}
+	mockFS := &FsClientMock{MkdirAllReturns: []error{nil}, WriteFileReturns: []error{nil}}
+	// Create a Daemon instance with mocked clients
+	d := Daemon{
```

Comment:
let's use the exported interface `NewClusterDrivenDaemon` directly, we'll be able to avoid making assumption on the object itself and avoid falling into stuff like #476

Reply:
thanks for this tip!

Reply:
The whole file should use that exported interface so ill update them all in another commit once I get this generally working :smile: